cp23/hpi-fun2/dinetravel/problem.tex

```diff
@@ -0,0 +1,52 @@
+\input{template.tex}
+
+\begin{document}
+
+\makeheader
+
+\placeholder{Problem description}
+
+One fateful night, the scientist Mabel McFly woke up from the most beautiful dream: Not only did she fly among the stars in a blue box, she also enjoyed an exotic alien dish named “fruit lasagna” that doesn’t exist in real life. Back in harsh reality, she refused to say goodbye to that godlike treat and decided to take matters into her own hands: She would use the time machine she designed to nudge the past in such a way that back in the present, fruit lasagnas will have been a traditional dish deeply intertwined with the foundations of culinary history.
+
+As everyone knows, time travel is a delicate thing. In order to precisely alter history for her dream dish, Mabel now wants to analyze the closest thing there is to the fruit lasagna in this world, the vegetable lasagna.
+As a thorough researcher, she consults many vegetable lasagna historians and notes all important events in a timeline. The historians also inform her about causal dependencies between those events, meaning some event A caused another event B.
+
+When Mabel evaluates all her notes, she realizes: She’s so close! She has uncovered almost the entire history of the vegetable lasagna’s origin. Only a single causal connection is missing and thereby prevents her from understanding the concept of vegetable lasagna on a more fundamental, holistic level.
+
+To estimate how much research still needs to be done, she wants to find the number of possibilities for the last connection so that it still conforms to the timeline given by the historians as well as the other known connections. She knows all the events that she noted directly or indirectly caused vegetable lasagna to emerge.
```

Comment:
I don't understand completely. So the inserted edge has to conform to the topological order. Is the last sentence supposed to say that every path should have a path to the lasagna? Can you clarify that in the statement?